tests/std/tests/P0980R1_constexpr_strings/test.cpp

```diff
@@ -0,0 +1,1758 @@
+        insert_const_char.insert(insert_const_char.cbegin() + 5, to_be_inserted);
+        assert(equalRanges(insert_const_char, "Hello, fluffy kittens"sv));
+
+        str insert_char_rvalue = get_literal_input<CharType>();
+        insert_char_rvalue.insert(insert_char_rvalue.begin() + 5, CharType{','});
+        assert(equalRanges(insert_char_rvalue, "Hello, fluffy kittens"sv));
+
+        str insert_const_char_rvalue = get_literal_input<CharType>();
+        insert_const_char_rvalue.insert(insert_const_char_rvalue.cbegin() + 5, CharType{','});
+        assert(equalRanges(insert_const_char_rvalue, "Hello, fluffy kittens"sv));
+
+        str insert_range(2, CharType{'b'});
+        const auto it = insert_range.insert(
+            insert_range.begin() + 1, begin(get_view_input<CharType>()), end(get_view_input<CharType>()));
+        assert(it == insert_range.begin() + 1);
+        assert(equalRanges(insert_range, "bHello fluffy kittensb"sv));
+
+        str insert_const_range(2, CharType{'b'});
+        const auto cit = insert_const_range.insert(
+            insert_const_range.cbegin() + 1, begin(get_view_input<CharType>()), end(get_view_input<CharType>()));
+        assert(cit == insert_const_range.cbegin() + 1);
+        assert(equalRanges(insert_const_range, "bHello fluffy kittensb"sv));
+
+        str insert_initializer = get_literal_input<CharType>();
+        const auto it_ilist    = insert_initializer.insert(insert_initializer.begin() + 6, {'c', 'u', 't', 'e', ' '});
+        assert(it_ilist == insert_initializer.begin() + 6);
+        assert(equalRanges(insert_initializer, "Hello cute fluffy kittens"sv));
+
+        str insert_const_initializer = get_literal_input<CharType>();
+        const auto cit_ilist =
+            insert_const_initializer.insert(insert_const_initializer.cbegin() + 6, {'c', 'u', 't', 'e', ' '});
+        assert(cit_ilist == insert_const_initializer.cbegin() + 6);
+        assert(equalRanges(insert_const_initializer, "Hello cute fluffy kittens"sv));
+
+        str insert_pos_str  = get_literal_input<CharType>();
+        const str to_insert = "cute and scratchy ";
+        insert_pos_str.insert(6, to_insert);
+        assert(equalRanges(insert_pos_str, "Hello cute and scratchy fluffy kittens"sv));
+
+        str insert_pos_substr = get_literal_input<CharType>();
+        insert_pos_substr.insert(6, to_insert, 0, 5);
+        assert(equalRanges(insert_pos_substr, "Hello cute fluffy kittens"sv));
+
+        const string_view_convertible<CharType> convertible;
+        str insert_pos_conversion = get_literal_input<CharType>();
+        insert_pos_conversion.insert(6, convertible);
+        assert(equalRanges(insert_pos_conversion, "Hello Hello fluffy kittensfluffy kittens"sv));
+
+        str insert_pos_conversion_substr = get_literal_input<CharType>();
+        insert_pos_conversion_substr.insert(6, convertible, 6, 7);
+        assert(equalRanges(insert_pos_conversion, "Hello fluffy fluffy kittens"));

```

Comment:
In MSVC-internal testing, I'm seeing an assertion failure on this line - and it is the first error emitted. I don't yet know what's happening here:

```
.\test.cpp(1734): error C2131: expression did not evaluate to a constant
.\test.cpp(592): note: failure was caused by call of undefined function or one not declared 'constexpr'
.\test.cpp(592): note: see usage of '_wassert'
```

Reply:
Oh! This one is just a typo! It's inspecting `insert_pos_conversion` from the *previous* block of tests. It should be inspecting `insert_pos_conversion_substr`.

Reply:
I'm bad at using the variable I actually intend to 😉 
```cpp
assert(equalRanges(insert_pos_conversion_substr.insert(6, convertible, 6, 7);
```

Reply:
Aha, there's a second typo here! After fixing the variable name, the assertion was *still* failing, because `"Hello fluffy fluffy kittens"` is a plain string literal, so when it's passed to `equalRanges()`, it's treated as an array containing a null terminator. All of the other assertions are using `"expected"sv` UDLs, so the `string_view`s don't contain null terminators.

Reply:
Will also push a fix with my next batch of changes.

Reply:
> Aha, there's a second typo here! After fixing the variable name, the assertion was _still_ failing, because `"Hello fluffy fluffy kittens"` is a plain string literal, so when it's passed to `equalRanges()`, it's treated as an array containing a null terminator.

FWIW, string literals have been a plague on Ranges in this manner forever. Everyone knows that `"abc"` should be a range with 3 elements, with the notable exception of the C++ type system =(